src/chessboard/index.tsx

```diff
@@ -25,53 +25,92 @@ export type ClearPremoves = {
   clearPremoves: (clearLastPieceColour?: boolean) => void;
 };
 
-export const Chessboard = forwardRef<ClearPremoves, ChessboardProps>((props, ref) => {
-  const { customDndBackend, customDndBackendOptions, ...otherProps } = props;
-  const [clientWindow, setClientWindow] = useState<Window>();
-  const [backendSet, setBackendSet] = useState(false);
-  const [isMobile, setIsMobile] = useState(false);
-  const [boardWidth, setBoardWidth] = useState(props.boardWidth);
+export const Chessboard = forwardRef<ClearPremoves, ChessboardProps>(
+  (props, ref) => {
+    const { customDndBackend, customDndBackendOptions, ...otherProps } = props;
+    const [clientWindow, setClientWindow] = useState<Window>();
+    const [backendSet, setBackendSet] = useState(false);
+    const [isMobile, setIsMobile] = useState(false);
+    const [boardWidth, setBoardWidth] = useState(props.boardWidth);
 
-  const boardRef = useRef<HTMLObjectElement>(null);
+    const boardRef = useRef<HTMLObjectElement>(null);
+    const boardContainerRef = useRef<HTMLDivElement>(null);
 
-  useEffect(() => {
-    setIsMobile("ontouchstart" in window);
-    setBackendSet(true);
-    setClientWindow(window);
-  }, []);
+    const [boardContainerPos, setBoardContainerPos] = useState({
+      left: 0,
+      top: 0,
+    });
 
-  useEffect(() => {
-    if (props.boardWidth === undefined && boardRef.current?.offsetWidth) {
-      const resizeObserver = new ResizeObserver(() => {
-        setBoardWidth(boardRef.current?.offsetWidth as number);
+    //Set BoardContainerPos initially when boardContainerRef is set. Position is used when dragOutsideOfBoard={false}
+    useEffect(() => {
+      if (boardContainerRef.current) {
+        const { left, top } = boardContainerRef.current.getBoundingClientRect();
+        setBoardContainerPos({ left, top });
+      }
+    }, [boardContainerRef.current]);
+
+    useEffect(() => {
+      window.addEventListener("resize", (e) => {
+        if (boardContainerRef.current) {
+          const { left, top } =
+            boardContainerRef.current.getBoundingClientRect();
+          setBoardContainerPos({ left, top });
+        }
+        return () => window.removeEventListener("resize", (e) => {});
```

Comment:
Just consider to use `resizeObserver` instead
```suggestion
```

Reply:
Hmm I cant seem to get it to handle resizing as reliably as with the event listener. Maybe I am missing something though. Here is an example of what I was trying with no success. 

`useEffect(() => {
      if (boardContainerRef.current) {
        const { left, top } = boardContainerRef.current.getBoundingClientRect();
        setBoardContainerPos({ left, top });

        const resizeObserver = new ResizeObserver(() => {
          if (boardContainerRef.current) {
            const { left, top } =
              boardContainerRef.current.getBoundingClientRect();
            setBoardContainerPos({ left, top });
          }
        });

        resizeObserver.observe(boardContainerRef.current);

        return () => {
          resizeObserver.disconnect();
        };
      }
    }, [boardContainerRef.current, window]);`

Reply:
We already use `ResizeObserver`
And I have shown how to extract data you need here https://github.com/Clariity/react-chessboard/pull/98/files#r1313796084

These useefects can be deleted